### ISSUE TITLE
Node.js v14 support

### ISSUE THREAD

**Issue by astorije** [vmware/clarity#5356]
Node.js v14 [is the currently Active LTS](https://nodejs.org/en/about/releases/). I have been running it locally for some time now, and did not experience any issue with it while working on #5353.
The only 2 things I have noticed:

- `engines.node` in `package.json` needs to be updated from `">=12 <13"` to `"^12 || ^14"`
- There is an error popping up when running `yarn install` related to `node-gyp`. We experienced this on our own projects too, and as far as we could tell, even though it caused an error, it did not actually have any impact. We ended up resolving/deduping a few dependencies in our `yarn.lock` files and the error went away (I'm not sure which ones anymore though, and probably irrelevant since your setup is different)

For reference, here is the error that pops up locally:

<details>

```
[5/5] 🔨  Building fresh packages...
[13/58] ⠐ @parcel/source-map
[12/58] ⠐ webpack-cli
[11/58] ⠐ core-js-bundle
[9/58] ⠐ node-sass
error /.../node_modules/node-pty: Command failed.
Exit code: 1
Command: node scripts/install.js
Arguments:
Directory: /.../node_modules/node-pty
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@14.9.0 | darwin | x64
gyp info spawn /usr/bin/python
gyp info spawn args [
gyp info spawn args   '/.../node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/.../node_modules/node-pty/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/.../node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/.../.node-gyp/14.9.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/.../.node-gyp/14.9.0',
gyp info spawn args   '-Dnode_gyp_dir=/.../node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/.../.node-gyp/14.9.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/.../node_modules/node-pty',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/.../node_modules/node-gyp/lib/configure.js:345:16)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Darwin 19.6.0
gyp ERR! command "/usr/local/Cellar/node/14.9.0/bin/node" "/.../node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /.../node_modules/node-pty
```

</details>

(I know the output talks about Xcode, but it does seem unrelated, and I do have Xcode correctly installed and set up)

**Comment by bdryanovski** [comment]
In addition to the possible update checklist, a change into the Github Actions is also required - to use a later version or we gonna have a mismatch between local and CI build env. 

Most of us use [nvm](https://github.com/nvm-sh/nvm) to handle the different versions of the node between projects and get automatically the requested version. 

Made a quick test and did not find any issues with updating to LTS.

Is there any reason for using Node LTS except the obvious inconvenience of rebuilding and switching nodes?

**Comment by astorije** [comment]
_EDIT: See comment below, this is incorrect..._

We use nvm too actually, regardless of the LTS/non-LTS or version used. This is less about _switching_ to a LTS and more about adding official support to Node v14, since the current version of Clarity is set up to explicitly disallow it (see `engines.node` in `package.json`).

Node 14 is the active LTS and Node 15 is the current version (in other words, those are the 2 versions immediately available on the homepage of https://nodejs.dev/). This ticket could be titled "Node.js v14 and v15 support" instead, but I find that in enterprise environment, people usually stick to LTSes versions more often, especially since v15 will be EOL in 6 months, so I think adding support for v14 is more important (though you're likely to support both for free).

If anyone were to start a new project and looking at [this list](https://nodejs.org/en/about/releases/) to figure out which version of Node to run, they'd see v12 is in maintenance mode and EOL in a little over a year, so they would likely run v14 instead, or v15 if they're ready for a quick EOL.

Does any of this answer what you were wondering?

**Comment by gnomeontherun** [comment]
We'll want to update to v14 sometime, but until v5 is out I don't want to see any more build changes. After v5, we could consider updating it with some of our build improvements.

**Comment by astorije** [comment]
_EDIT: See comment below, this is incorrect..._

@gnomeontherun, I'm sorry if I'm not expressing myself clearly, I'm not talking about changing your build or updating anything, but adding support for Node v14 at the consumer level.
In other words, if a developer wants to build a Clarity project with Node v14 on their machine, they will get errors and warnings which I believe could be resolved without changing any code or internal build configuration to Clarity.

Any chance v5 could support Node v14? I'm happy to try to help with a PR if so.

**Comment by astorije** [comment]
Oh wow. This is what happens when I resume after the holidays, being mistaken about my own issue 🤦
Sorry I'm an idiot, ahah. For some reason I was convincing myself I opened this issue after using Clarity on our projects, but no I opened it while _building Clarity_.
Me ➡️ 🚪

To whoever stops by this ticket, nothing prevents you to use Clarity on a project built with Node 14: https://unpkg.com/@cds/core@5.0.0-next.0/package.json (i.e. `engines.node` is not specified here). In fact, I've been doing this for a while without any issue myself 🙄
